File: Python/0860 Lemonade Change.py
class Solution(object):
    def lemonadeChange(self, bills):
        """
        :type bills: List[int]
        :rtype: bool
        """
        dic = {5: 0, 10: 0, 20: 0}
        for i in bills:
            if i == 5:
                dic[5] += 1
            elif i == 10 and dic[5] >= 1:
                dic[5] -= 1
                dic[10] += 1
            elif i == 10 and dic[5] < 1:
                return False
            elif dic[10] >= 1 and dic[5] >= 1:
                dic[10] -= 1
                dic[5] -= 1
                dic[20] += 1
            elif dic[5] >= 3:
                dic[5] -= 3
                dic[20] += 1
            else:
                return False
        return True
            

File: C&C++/0860 Lemonade Change.cpp
bool lemonadeChange(int* bills, int billsSize){
    int f5=0,f10=0,f20=0;
    for(int i=0;i<billsSize;i++){
        if(bills[i]==5) f5++;
        if(bills[i]==10){
            f10++;
            if(f5>=1) f5--;
            else return false;
        }
        if(bills[i]==20){
            f20++;
            if(f5>0 && f10>0) f5--,f10--;
            else if(f5>=3) f5-=3;
            else return false;
        }
    }
    return true;
}

